instrumented/integration/src/androidTest/kotlin/com/datadog/android/sdk/integrationtests/profiling/BatteryProfileForLogs.kt

```diff
@@ -0,0 +1,52 @@
+package com.datadog.android.sdk.integrationtests.profiling
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import com.datadog.android.sdk.integrationtests.ActivityProfiling
+import com.datadog.android.sdk.integrationtests.utils.BatteryProfilingRule
+import com.datadog.android.sdk.integrationtests.utils.MockServerRule
+import java.io.IOException
+import java.util.concurrent.CountDownLatch
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+internal class BatteryProfileForLogs {
+
+    @get:Rule
+    val mockServerRule = MockServerRule(ActivityProfiling::class.java)
+    @get:Rule
+    val batteryProfilingRule = BatteryProfilingRule()
+
+    @Test
+    fun profileCrashLogWithLargeNumberOfAttributes() {
+        val crash = IOException()
+        val attributes = mutableMapOf<String, String>()
+        for (i in 0..100) {
+            attributes["key$i"] = "value$i"
+        }
+        val countDownLatch = CountDownLatch(120)
+
+        batteryProfilingRule.profile({
+            InstrumentationRegistry.getInstrumentation().runOnMainSync {
+                // we are going to simulate 1000 logs per minute => ~ 8 logs every 500 ms
+                while (countDownLatch.count > 0) {
```

Comment:
I'm not comfortable having all those magic number to generate the 1000 logs / minute thing. Maybe we could create a function like to generically take a duration, number of occurences and repeat the appropriate number of time, with an appropriate sleep duration? Something like : 
```kotlin
fun repeat(
    occurences : Int,
    durationMs : Long,
    minimumSleepMs: Long,
    block: () -> Unit
) {
    val periodMs = durationMs / occurences
    val sleepMs = max(minimumSleepMs, periodMs)
    val repeatPerIteration = sleepMs / periodMs
    val iterations = durationMs / sleepMs

    for (i in 0 ..iterations) {
        repeat(repeatPerIteration.toInt()) { block() }
        Thread.sleep(sleepMs)
    }
}
```

Reply:
Yup, was actually thinking to something like that, tks for the idea !!